tests/mocha/test_helpers/common.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.declareModuleId('Blockly.test.helpers.common');
+
+/**
+ * Test case configuration.
+ * @record
+ */
+export class TestCase {
+  /**
+   * Class for a test case configuration.
+   */
+  constructor() {
+    /**
+     * @type {string} The title for the test case.
+     */
+    this.title;
+    /**
+     * @type {boolean|undefined} Whether this test case should be skipped.
+     *   Used to skip buggy test case and should have an associated bug.
+     */
+    this.skip;
+    /**
+     * @type {boolean|undefined} Whether this test case should be called as
+     *   only. Used for debugging.
+     */
+    this.only;
+  }
+}
+
+/**
+ * Test suite configuration.
+ * @record
+ * @template {TestCase} T
+ * @template {TestSuite} U
```

Comment:
I note that this is causing additional warnings during build (specifically, while doing `build:deps`:
```
WARNING in /Users/cpcallen/src/blockly/tests/mocha/test_helpers/common.js at 38, 3: Bounded generic semantics are currently still in development
WARNING in /Users/cpcallen/src/blockly/tests/mocha/test_helpers/common.js at 39, 3: Bounded generic semantics are currently still in development
```
(These two plus three more below).

I checked to see if there was some way to suppress these warnings, but I was not able to find one and filed google/closure-library#1156 to request such a feature.